src/plugins/condition/criterion/AllTelemetryCriterion.js

```diff
@@ -159,6 +160,34 @@ export default class AllTelemetryCriterion extends TelemetryCriterion {
             });
     }
 
+    getDescription() {
+        const telemetryDescription = this.telemetry === 'all' ? 'all telemetry' : 'any telemetry';
+        let metadataValue = this.metadata;
+        let inputValue = this.input;
+        if (this.metadata) {
+            const telemetryObjects = Object.values(this.telemetryObjects);
+            for (let i=0; i < telemetryObjects.length; i++) {
+                const telemetryObject = telemetryObjects[i];
+                const telemetryMetadata = this.openmct.telemetry.getMetadata(telemetryObject);
+
+                const metadataObj = telemetryMetadata.valueMetadatas.find((metadata) => metadata.key === this.metadata);
+                if (metadataObj) {
```

Comment:
This section of code appears to be shared with its parent, TelemetryCriterion. Might also help the readability of `getDescription()` which is pretty long.

Reply:
Done. Refactored code for reuse.